installer/windows/Product.wxs

```diff
@@ -57,9 +57,18 @@
           Account="[SERVICEACCOUNT]"
           Password="[SERVICEPASSWORD]"
           ErrorControl="normal"
-          Arguments="run --config-dir &quot;[INSTALLFOLDER]\&quot; --identity-dir &quot;[IDENTITYDIR]\&quot; --binary-location &quot;[INSTALLFOLDER]\storagenode.exe&quot; --log &quot;[INSTALLFOLDER]\storagenode-updater.log&quot;"
-          />
-        <ServiceControl Id="StoragenodeUpdaterStartService" Start="install" Stop="both" Remove="uninstall" Name="storagenode-updater" Wait="yes" />
+          Arguments="run --config-dir &quot;[INSTALLFOLDER]\&quot; --identity-dir &quot;[IDENTITYDIR]\&quot; --binary-location &quot;[INSTALLFOLDER]\storagenode.exe&quot; --log &quot;[INSTALLFOLDER]\storagenode-updater.log&quot;">
+            <util:ServiceConfig
+              ServiceName="storagenode-updater"
+              ResetPeriodInDays="1"
+              RestartServiceDelayInSeconds="60"
+              FirstFailureActionType="restart"
+              SecondFailureActionType="runCommand"
+              ThirdFailureActionType="runCommand"
+              ProgramCommandLine="cmd /c &quot;(echo %date% %time% recovery triggered to execute in 3 minutes &amp;&amp; timeout 180 >NUL &amp;&amp; echo %date% %time% recovery command executed: &amp;&amp; copy /y &quot;[INSTALLFOLDER]storagenode-updater.old.exe&quot; &quot;[INSTALLFOLDER]storagenode-updater.exe&quot; &amp;&amp; net start storagenode-updater) >> &quot;[INSTALLFOLDER]storagenode-updater.log&quot;&quot;"
```

Comment:
Can we add `2>&1` to redirect stderr to the log file too? Otherwise, we cannot see errors.

i.e. `... ) >> &quot;[INSTALLFOLDER]storagenode-updater.log&quot; 2>&1&quot;`

Reply:
I think so, let me try this

Reply:
I also found why we miss the log messages from storagenode-updater after recovery. The root cause of the issue is that we redirect the output of the recovery command to the same file. I am not sure why this suppresses the output of the storagenode-updater process to the same file, but we can resolve this by redirecting the output of the recovery to command to a different file, e.g. `storagenode-updater-recovery.log`.

So the final change to the recovery command should look like this:

```
... ) >> &quot;[INSTALLFOLDER]storagenode-updater-recovery.log&quot; 2>&amp;1&quot;
```

Reply:
fixed, thanks for help!